packages/visual-editor/src/utils/resolveTranslatableString.tsx

```diff
@@ -34,33 +34,48 @@ export const resolveTranslatableString = (
 };
 
 /**
- * Converts a type TranslatableRichText to a type that can be viewed on the page
+ * Converts a type TranslatableRichText to string or RTF Element
  * @param translatableRichText
  * @param locale
  */
 export const resolveTranslatableRichText = (
   translatableRichText?: TranslatableRichText,
   locale: string = "en"
 ): string | React.ReactElement => {
-  if (!translatableRichText) return "";
+  try {
+    if (!translatableRichText) return "";
+
+    if (
```

Comment:
There's this if condition, the one below, and then the logic in isRichText which also checks if the type is an object like on line 55. Can this be simplified to just check if translatableRichText is a map, grab the locale, and then always call toStringOrElement which happens in both of these conditions?